core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -190,15 +189,14 @@ class ReplicaStateMachine(controller: KafkaController) extends Logging {
                                                 targetState))
             case _ =>
               // check if the leader for this partition ever existed
-              controllerContext.partitionLeadershipInfo.get(topicAndPartition) match {
-                case Some(leaderIsrAndControllerEpoch) =>
-                  brokerRequestBatch.addLeaderAndIsrRequestForBrokers(List(replicaId), topic, partition, leaderIsrAndControllerEpoch,
-                    replicaAssignment)
-                  replicaState.put(partitionAndReplica, OnlineReplica)
-                  stateChangeLogger.trace("Controller %d epoch %d changed state of replica %d for partition %s from %s to %s"
-                    .format(controllerId, controller.epoch, replicaId, topicAndPartition, currState, targetState))
-                case None => // that means the partition was never in OnlinePartition state, this means the broker never
-                  // started a log for that partition and does not have a high watermark value for this partition
+              // None means the partition was never in OnlinePartition state, this means the broker never
```

Comment:
This probably comes down to personal preference, but in cases when we need to communicate a non-trivial explanation for the `None` case, I found the pattern matching approach easier to follow because the comment appears in the context of the case it applies to.